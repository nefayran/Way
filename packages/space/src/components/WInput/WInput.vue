<template>
  <div :class="[classes.wrapper]">
    <input
      :value="props.modelValue"
      :class="[classes.text]"
      type="text"
      @input="(event) => $emit('update:modelValue', event.target.value)"
    />
  </div>
</template>

<script lang="ts" setup>
import { withDefaults, computed, ref } from "vue";

// const modelValue = ref("");

interface Props {
  color?: string;
  fill?: string;
  size?: string;
  modelValue: string;
}

const props = withDefaults(defineProps<Props>(), {
  color: "primary--base",
  fill: "secondary--base",
  size: "lg",
});

const emit = defineEmits(["update:modelValue"]);

const classes = computed(() => {
  return {
    wrapper: `w-input w-input--${props.size} ${props.fill}`,
    text: `w-text font-primary text-${props.color}`,
  };
});
</script>

<style lang="scss">
/* === Input Sizes === */
$sizes: (xl, lg, md, sm, xs) !default;
.w-input {
  border: none;

  @each $size in $sizes {
    &--#{$size} {
      height: var(--height-#{$size});
      min-height: var(--height-#{$size});
      padding-left: var(--space-#{$size});
      padding-right: var(--space-#{$size});
      border-radius: var(--border-radius-#{$size});
      min-width: var(--width-#{$size});

      .w-text {
        width: 100%;
        height: 100%;
        border: none;
        background: transparent;
        font-size: var(--font-size-#{$size});

        &:focus {
          outline: none;
        }
      }
    }
  }
}
</style>
