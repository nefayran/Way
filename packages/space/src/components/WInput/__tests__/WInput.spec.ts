import { mount } from "@vue/test-utils";
import WInput from "../WInput.vue";

describe("WInput.vue", () => {
  test("should have an w-input class", () => {
    const wrapper = mount(WInput, {
      propsData: {
        value: "test",
      },
    });
    expect(wrapper.classes()).toContain("w-input");
    expect(wrapper.html()).toMatchSnapshot();
  });
});
