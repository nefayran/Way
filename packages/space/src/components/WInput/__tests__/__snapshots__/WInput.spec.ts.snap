// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WInput.vue should have an w-input class 1`] = `<div class="w-input w-input--lg secondary--base"><input class="w-text font-primary text-primary--base" type="text"></div>`;
